docs/USING_PUBLIC_CLOUDS.rst

```diff
@@ -0,0 +1,459 @@
+   section of this document.
+
+AWS S3 as a backend
+-------------------
+
+From the AWS S3 Console
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Create a bucket where you will host your data for this new location constraint.
+This bucket must have versioning enabled. This is an option you may choose to
+activate at step 2 of Bucket Creation in the console.
+In this example, our bucket will be name ``zenkobucket`` and has versioning
+enabled.
+
+From the CloudServer repository
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+locationConfig.json
+^^^^^^^^^^^^^^^^^^^
+
+Edit this file to add a new region, which should describe the AWS S3 hosted
+bucket you'll be writing your data to. There are a few configurable options here:
+
+- :code:`type` : set to :code:`aws_s3` to indicate this region is hosted in AWS S3;
+- :code:`legacyAwsBehavior` : set to :code:`true` to indicate this region should
+  behave like AWS S3 :code:`us-east-1` region, set to :code:`false` to indicate
+  this region should behave like any other AWS S3 region;
+- :code:`bucketName` : set to an *existing bucket* in your AWS S3 Account; this
+  is the bucket that will host all the data written in this region;
+- :code:`awsEndpoint` : set to your bucket's endpoint, usually :code:`s3.amazonaws.com`;
+- :code:`bucketMatch` : set to :code:`true` if you want your object name to be the
+  same in your local bucket and your AWS S3 bucket; set to :code:`false` if you
+  want your object name to be of the form :code:`{{localBucketName}}/{{objectname}}`
+  in your AWS S3 hosted bucket;
+- :code:`credentialsProfile` and :code:`credentials` are two ways to provide
+  your AWS S3 credentials for that bucket, *use only one of them* :
+  - :code:`credentialsProfile` : set to the profile name allowing you to access
+    your AWS S3 bucket from your :code:`~/.aws/credentials` file;
+  - :code:`credentials` : set the two fields inside the object (:code:`accessKey`
+    and :code:`secretKey`) to their respective values from your AWS credentials.
+
+.. code:: json
+
+    (...)
+    "aws-test": {
+        "type": "aws_s3",
+        "legacyAwsBehavior": true,
+        "details": {
+            "awsEndpoint": "s3.amazonaws.com",
+            "bucketName": "zenkobucket",
+            "bucketMatch": true,
+            "credentialsProfile": "zenko"
+        }
+    },
+    (...)
+
+.. code:: json
+
+    (...)
+    "aws-test": {
+        "type": "aws_s3",
+        "legacyAwsBehavior": true,
+        "details": {
+            "awsEndpoint": "s3.amazonaws.com",
+            "bucketName": "zenkobucket",
+            "bucketMatch": true,
+            "credentials": {
+                "accessKey": "WHDBFKILOSDDVF78NPMQ",
+                "secretKey": "87hdfGCvDS+YYzefKLnjjZEYstOIuIjs/2X72eET"
+            }
+        }
+    },
+    (...)
+
+.. WARNING::
+   If you set :code:`bucketMatch` to :code:`true`, we strongly advise for having
+   only one local bucket per AWS S3 location. Indeed, since your objects names
```

Comment:
Without bucketMatch set to false, your object names in your AWS S3 bucket will not be prefixed with your Cloud Server bucket name. This means that if you put an object `foo` to your CloudServer bucket `zenko1` and you then put a different `foo` to your CloudServer bucket `zenko2` and both `zenko1` and `zenko2` point to the same AWS bucket, the second `foo` will overwrite the first `foo`.

Reply:
changed